Core/src/Testing/TestHelpers.php

```diff
@@ -256,6 +257,26 @@ public static function getPrivateProperty($class, $property)
         return $c($class);
     }
 
+    /**
+     * Compare actual and expected results when the requirement for comparision
+     * is `===` including outlier cases like:
+     *  - Comparing NAN
+     *  - Comparing floating point values with some delta
+     */
+    public static function compareResult($expected, $actual)
+    {
+        if (is_float($expected)) {
+            if (is_nan($expected)) {
+                Assert::assertNan($actual);
+            } else {
+                Assert::assertEqualsWithDelta($expected, $actual, 0.01);
+            }
+        } else {
+            // Used because assertEquals(null, '') doesn't fails
+            Assert::assertSame($expected, $actual);
+        }
+    }
```

Comment:
I'm not a huge fan of static calls. I think we would be better off putting this in [`Google\Cloud\Core\Testing\System\SystemTestCase`](https://github.com/googleapis/google-cloud-php/blob/main/Core/src/Testing/System/SystemTestCase.php), which is extended by Firestore and Datastore tests? This way we could override the function, and also not have to use static calls to `TestHelpers`. This has the additional benefit that we can also replace the static calls to `Assert` with standard `TestCase` functions:

```php
$this->assertNan($actual);
$this->assertEqualsWithDelta($expected, $actual, 0.01);
$this->assertSame($expected, $actual);
```

Reply:
We heavily need them for unit testing as they are separated in files much more than system tests.